lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/GeneratingSuggester.java

```diff
@@ -152,7 +153,7 @@ private void filterSuitableEntries(String word, IntsRef forms, List<Root<String>
       for (String guess : expandRoot(weighted.word, misspelled)) {
         String lower = dictionary.toLowerCase(guess);
         int sc =
-            ngram(misspelled.length(), misspelled, lower, EnumSet.of(NGramOptions.ANY_MISMATCH))
+            anyMismatchNgram(misspelled.length(), misspelled, lower, false)
```

Comment:
`ANY_MISMATCH` is now baked into the method's name, `false` stands for lack of `WEIGHTED` option